Modules/_testexternalinspection.c

```diff
@@ -0,0 +1,621 @@
+#define _GNU_SOURCE
+
+#ifdef __linux__
+#    include <elf.h>
+#    include <sys/uio.h>
+#    if INTPTR_MAX == INT64_MAX
+#        define Elf_Ehdr Elf64_Ehdr
+#        define Elf_Shdr Elf64_Shdr
+#        define Elf_Phdr Elf64_Phdr
+#    else
+#        define Elf_Ehdr Elf32_Ehdr
+#        define Elf_Shdr Elf32_Shdr
+#        define Elf_Phdr Elf32_Phdr
+#    endif
+#endif
+
+#ifdef __APPLE__
+#    include <libproc.h>
+#    include <mach-o/fat.h>
+#    include <mach-o/loader.h>
+#    include <mach-o/nlist.h>
+#    include <mach/mach.h>
+#    include <mach/mach_vm.h>
+#    include <mach/machine.h>
+#    include <sys/proc.h>
+#    include <sys/sysctl.h>
+#endif
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/param.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <execinfo.h>
+
+
+#ifndef Py_BUILD_CORE_BUILTIN
+#    define Py_BUILD_CORE_MODULE 1
+#endif
+#include "Python.h"
+#include <internal/pycore_runtime.h>
+
+#ifndef HAVE_PROCESS_VM_READV
+#    define HAVE_PROCESS_VM_READV 0
+#endif
+
+#ifdef __APPLE__
+
+static void*
+analyze_macho64(mach_port_t proc_ref, void* base, void* map)
+{
+    struct mach_header_64* hdr = (struct mach_header_64*)map;
+    int ncmds = hdr->ncmds;
+
+    int cmd_cnt = 0;
+    struct segment_command_64* cmd = map + sizeof(struct mach_header_64);
+
+    mach_vm_size_t size = 0;
+    mach_msg_type_number_t count = sizeof(vm_region_basic_info_data_64_t);
+    mach_vm_address_t address = (mach_vm_address_t)base;
+    vm_region_basic_info_data_64_t region_info;
+    mach_port_t object_name;
+
+    for (register int i = 0; cmd_cnt < 2 && i < ncmds; i++) {
+        if (cmd->cmd == LC_SEGMENT_64 && strcmp(cmd->segname, "__DATA") == 0) {
+            while (cmd->filesize != size) {
+                address += size;
+                if (mach_vm_region(
```

Comment:
AFAIK, `mach_vm_region` was introduced with macOS 10.4 and support for macOS 10.3 has not been explicitly dropped, so we should add a configure check and a test guard for this API. cc. @ronaldoussoren

Reply:
Makes sense, I still need to guard against WASI, FreeBSD and I am fighting the 32 bit Raspbian builedbot for now, so don't take the current version too seriously

Reply:
Thanks for the catch btw

Reply:
10.4 is IMHO old enough to no longer bother with a configure check for this API. We've already rejected requests to add workarounds for pre-10.9 macOS versions due to added complexity.

Reply:
> I still need to guard against WASI [...]

FTR IIRC, `mmap` is unavailable on WASI.

Reply:
> 10.4 is IMHO old enough to no longer bother with a configure check for this API. We've already rejected requests to add workarounds for pre-10.9 macOS versions due to added complexity.

Ok, thanks for the info. We should consider updating PEP-11 with such information.

Reply:
> > 10.4 is IMHO old enough to no longer bother with a configure check for this API. We've already rejected requests to add workarounds for pre-10.9 macOS versions due to added complexity.
> 
> Ok, thanks for the info. We should consider updating [PEP-11](https://peps.python.org/11) with such information.

Indeed, I'll see if I can start a thread about this on d.p.o. in the near future.

Reply:
@ronaldoussoren Are you ok if we add a configure check + a guard against this API? Or is there a cleaner way to check for this at compile time?

Reply:
Oh sorry, race condition in my GitHub UI :(